pkgs/servers/home-assistant/custom-components/sleep_as_android/package.nix

```diff
@@ -0,0 +1,68 @@
+{
+  lib,
+  buildHomeAssistantComponent,
+  fetchFromGitHub,
+  pyhaversion,
+  nix-update-script,
+  # Test dependencies
+  pytestCheckHook,
+  pytest-homeassistant-custom-component,
+  aiohttp-cors,
+  paho-mqtt,
+}:
+let
+  version = "2.3.2";
+in
+buildHomeAssistantComponent {
+  owner = "IATkachenko";
+  domain = "sleep_as_android";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "IATkachenko";
+    repo = "HA-SleepAsAndroid";
+    tag = "v${version}";
+    hash = "sha256-aJKjHZcRdmiXJdtWRY4fv5oxCHTDIVpvZEwhIE9ISv8=";
+  };
+
+  dependencies = [
+    pyhaversion
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-homeassistant-custom-component
+    aiohttp-cors
+    paho-mqtt
+    # This is specified in the repos requirements.txt, but it breaks the build
+    # Given that it's about git, we should be able to ignore it
+    # pre-commit
+  ];
+
+  pytestFlagsArray = [
+    # Fixes `AttributeError: 'async_generator' object has no attribute 'data'`
+    "--asyncio-mode=auto"
+  ];
+
+  disabledTests = [
+    # Broken on upstream
+    # See https://github.com/home-assistant/developers.home-assistant/blob/ae84fbf6bf2bfcd978397c5db6850b1aca1b87a4/blog/2022-06-07-dataflow-constants-deprecation.md
+    #  AttributeError: module 'homeassistant.data_entry_flow' has no attribute 'RESULT_TYPE_FORM'
+    "tests/functional/test_config_flow.py"
```

Comment:
`disabledTests` is for disabling individual tests, not paths. I'm not sure if this has the effect you are imaginging it to have.